File: 898-transpose-matrix/solution.py
class Solution:
    def transpose(self, A):
        """
        :type A: List[List[int]]
        :rtype: List[List[int]]
        """
        r, c = len(A[0]), len(A)
        result = [[0]*c for _ in range(r)]
        for i in range(len(A)):
            for j in range(len(A[0])):
                result[j][i] = A[i][j]
        return result

File: 898-transpose-matrix/solution.cpp
class Solution {
public:
    vector<vector<int>> transpose(vector<vector<int>>& A) {
        vector<int> col(A.size());
        vector<vector<int>> result(A[0].size(), col);
        for (int i = 0; i < A.size(); i ++) {
            for (int j = 0; j < A[0].size(); j ++) {
                result[j][i] = A[i][j];
            }
        }
        return result;
    }
};
